src/linker/Linker.Steps/CodeRewriterStep.cs

```diff
@@ -43,6 +50,35 @@ void ProcessMethod (MethodDefinition method)
 				RewriteBodyToFalse (method);
 				break;
 			}
+
+			if (Annotations.IsReflected (method))
+				AddReflectedAttr (method);
+		}
+
+		void AddReflectedAttr (MethodDefinition method)
+		{
+			// Public methods have non-visible call sites by default, this attribute doesn't
+			// help us at all.
+			//
+			// See mono_aot_can_specialize in aot-compiler.c in mono
+			if (!method.IsPrivate)
+				return;
+
+			if (noOptAttr == null) {
+				var methodImpl = typeof (System.Runtime.CompilerServices.MethodImplAttribute);
```

Comment:
This is wrong, you need to use target reference type. Use `FindPredefinedType` instead

Reply:
I gave that a try before and Cecil reported that it couldn't import the MethodImplAttribute class when doing some emitter pass. I'll look back into it.

Reply:
Still running into cecil issues, cecil doesn't see any of the constructors for that type beyond the copy constructor.

Reply:
Also this: 

https://github.com/mono/linker/blob/fa9ccbdaf6907c69ef1bb117906f8f012218d57f/src/linker/Linker.Steps/CodeRewriterStep.cs#L51

doesn't actually change the emitted method's attributes at all.

Reply:
It's also definitely not tested, because if I remove those lines the tests pass.

Reply:
Fyi, the test infrastructure does not assert `ImplAttributes` currently.  See `VerifyMethodKept.VerifyMethodKept`.  We should hook up the same pattern used for `VerifyPseudoAttributes`